#
# Fixes opening a irc:// URL via "xchat -a --url=abc123" command while
# "Skip serverlist on startup" is off. This fix is only relevant when
# opening an initial instance of xchat (i.e not via dbus).
#
--- src/common/xchat.c	2006-01-02 22:38:23.000000000 +1100
+++ src/common/xchat.c	2006-03-27 23:08:18.000000000 +1100
@@ -918,7 +918,8 @@
 
 	servlist_init ();							/* load server list */
 
-	if (!prefs.slist_skip)
+	/* if we got a URL, don't open the server list GUI */
+	if (!prefs.slist_skip && !arg_url)
 		fe_serverlist_open (NULL);
 
 	/* turned OFF via -a arg */
@@ -937,7 +938,7 @@
 		}
 	} else
 	{
-		if (prefs.slist_skip)
+		if (prefs.slist_skip || arg_url)
 			new_ircwindow (NULL, NULL, SESS_SERVER, 0);
 	}
 }
